### PR TITLE
Eliminate the word 'view' from arcs.js

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: runtime/arc.js [hunk 1]
```diff
@@ -140,18 +138,18 @@ class Arc {
...
+      if (type.isCollection)
         type = type.primitiveType();
       if (type.isInterface) {
         interfaces += type.interfaceShape.toString() + '\n';
       }
       let key = this._storageProviderFactory.parseStringAsKey(handle.storageKey);
       switch (key.protocol) {
         case 'firebase':
-          handles += `view View${id++} of ${handle.type.toString()} '${handle.id}' @${handle._version} at '${handle.storageKey}'\n`;
+          handles += `store Store${id++} of ${handle.type.toString()} '${handle.id}' @${handle._version} at '${handle.storageKey}'\n`;
```
Comment (shaper) [inline runtime/arc.js]: I think you don't need the `.toString()` on `handle.type`, it's implicit.

File: runtime/arc.js [hunk 2]
```diff
@@ -171,7 +169,7 @@ class Arc {
           let data = JSON.stringify(serializedData);
           resources += data.split('\n').map(line => indent + line).join('\n');
           resources += '\n';
-          handles += `view View${id} of ${handle.type.toString()} '${handle.id}' @${handle._version} in View${id++}Resource\n`;
+          handles += `store Store${id} of ${handle.type.toString()} '${handle.id}' @${handle._version} in Store${id++}Resource\n`;
```
Comment (shaper) [inline runtime/arc.js]: Ah, I didn't know about `store`.

File: runtime/manifest-parser.peg [hunk 3]
```diff
@@ -449,7 +449,7 @@ ParticleProvidedSlotItem
   / ParticleProvidedSlotView
 
 ParticleProvidedSlotView
-  = 'view' whiteSpace view:lowerIdent eolWhiteSpace
+  = ('view' / 'handle') whiteSpace view:lowerIdent eolWhiteSpace
```
Comment (shaper) [inline runtime/manifest-parser.peg]: I wondered about perhaps replacing all internal-only `view` references while we're here but fine to do later.

File: runtime/type.js [hunk 4]
```diff
@@ -79,6 +79,16 @@ class Type {
     return this.variable;
   }
 
+  // TODO: rename SetView to Collection
+  // Once everything's moved over to this, we can change the
+  // underlying representation
+  get isCollection() {
+    return this.isSetView;
```
Comment (shaper) [inline runtime/type.js]: Also wondered about adding a deprecation note to the not-collection method names in some manner, but that would probably produce extreme spew, so instead am just mentioning here.

File: runtime/manifest-parser.peg (hunk 3 above)
Comment (shans) [reply runtime/manifest-parser.peg]: Yeah, we should definitely do it. I think I'll keep poking away at these smallish cleanup CLs until everything's good.

File: runtime/type.js (hunk 4 above)
Comment (shans) [reply runtime/type.js]: I discussed a less verbose deprecation with Scott this week. Once we figure something out I'll start to deprecate the old names.

File: runtime/arc.js (hunk 1 above)
Comment (shans) [reply runtime/arc.js]: I think I'd rather keep it though, for clarity about what's happening.

**Pull request merged**